modules/raft/src/main/java/org/apache/ignite/internal/raft/server/impl/JraftServerImpl.java

```diff
@@ -489,9 +491,17 @@ public boolean hasNext() {
                     public CommandClosure<WriteCommand> next() {
                         @Nullable CommandClosure<WriteCommand> done = (CommandClosure<WriteCommand>) iter.done();
                         ByteBuffer data = iter.getData();
-                        WriteCommand command = JDKMarshaller.DEFAULT.unmarshall(data.array());
+
+                        WriteCommand command = done == null ? JDKMarshaller.DEFAULT.unmarshall(data.array()) : done.command();
+
+                        long index = iter.getIndex();
 
                         return new CommandClosure<>() {
+                            @Override
+                            public long index() {
+                                return index;
+                            }
```

Comment:
```suggestion
                        return new CommandClosure<>() {
                             /** {@inheritDoc} */
                            @Override
                            public long index() {
                                return index;
                            }
```

Reply:
Other method in this class don't have "inheritDoc" comment, should I break consistency?